build/helper/codegen_helper.py

```diff
@@ -147,14 +154,26 @@ def _get_output_param_return_snippet(output_parameter, parameters, config):
 
 
 # TODO(marcoskirsch): Retrofit to call filter_parameters(function, parameter_usage_options)
-def get_method_return_snippet(parameters, config, use_numpy_array=False):
-    '''Returns a string suitable to use as the return argument of a Session method, i.e. "return reading_ctype.value"'''
+def get_library_interpreter_method_return_snippet(parameters, config, use_numpy_array=False):
+    '''Returns a string suitable to use as the return argument of a LibraryInterpreter method, i.e. "return reading_ctype.value"'''
     snippets = []
     for x in parameters:
         if x['direction'] == 'out' or x['size']['mechanism'] == 'ivi-dance':
             if x['numpy'] is False or use_numpy_array is False:
```

Comment:
@marcoskirsch Is this `or use_numpy_array is False` actually providing any value?
After going through the metadata and template that call this, I don't understand the argument's intended use.

Reply:
It does. use_numpy_array is passed in by the numpy templates. The functions that have a numpy variant also have a non-numpy variant. When invoked as the non-numpy variant, the "numpy" outputs need to be included. But when invoked as the numpy variant, these outputs are inputs instead, so they're excluded here.

Reply:
Excerpt from NI-SCOPE "FetchBinary8" function metadata.

```
            {
                'direction': 'out',
                'documentation': {
                    'description': '\nReturns an array whose length is the **numSamples** times number of\nwaveforms. Call niScope_ActualNumwfms to determine the number of\nwaveforms.\n\nNI-SCOPE returns this data sequentially, so all record 0 waveforms are\nfirst. For example, with a channel list of 0,1, you would have the\nfollowing index values:\n\nindex 0 = record 0, channel 0\n\nindex *x* = record 0, channel 1\n\nindex 2\\ *x* = record 1, channel 0\n\nindex 3\\ *x* = record 1, channel 1\n\nWhere *x* = the record length\n'
                },
                'name': 'waveform',
                'numpy': True,
                'size': {
                    'mechanism': 'python-code',
                    'value': '(num_samples * self.actual_num_wfms(channel_list))'
                },
                'type': 'ViInt8[]',
                'use_array': True
            },
```
I don't think I see waveform returned anywhere, although the documentation for _fetch_into functions seem to list it anyway.

Reply:
Yes, the fact that it's listed in the docstring is a pre-existing bug. And I have a fix for it...

* Yesterday I refactored the aforementioned TODO, wherein we should be using ParameterUsageOptions and filter_parameters instead of manually filtering the parameters. You can see that refactor [here](https://github.com/DavidCurtiss/nimi-python/commits/refactor-output-params), in the Sep 28 changes. I'll post a PR for that once this one is complete. (Unless you think I should go ahead and add it to this one, since I'm still blocked on merging this one anyway.)
.
[One of the changes](https://github.com/DavidCurtiss/nimi-python/commit/467a76c830bfe85991992010868d2e16abc61b38) in that refactor adds a API_NUMPY_OUTPUT_PARAMETERS option, which I then use in `documentation_helper.get_function_docstring` for numpy functions, which fixes ni-scope's docstrings.

Reply:
No, please no more big changes in this PR. My brain might crash.